sympy/stats/rv.py

```diff
@@ -742,7 +758,7 @@ def doit(self, evaluate=True, **kwargs):
             return Lambda(x, DiracDelta(x - expr))
         if (isinstance(expr, RandomSymbol) and
             hasattr(expr.pspace, 'distribution') and
-            isinstance(pspace(expr), SinglePSpace)):
+            isinstance(pspace(expr), (SinglePSpace, JointPSpace))):
```

Comment:
superclass?

Reply:
`JointPSpace` inherits from `ProductPSpace`, which has to be used if it is not an instance of either `SinglePSpace` or `JointPSpace`. These both inherit from `PSpace`, but `ProductPSpace` also inherits from it, so it would be an issue.

Reply:
maybe a common superclass `PSpaceWithDistribution`?

Reply:
@Upabjojr With the current implementation of `SinglePSpace` and `JointPSpace`, the only method the superclass would have will be `distribution`. I don't the a super class is needed.